lib/api/src/backend/sys/entities/engine.rs

```diff
@@ -184,6 +184,53 @@ impl NativeEngineExt for crate::engine::Engine {
     }
 }
 
+/// Trait for getting supported features from a backend engine
+pub trait EngineFeatures {
+    /// Gets the default and supported features for a given target
+    fn target_features_for_target(&self, target: &Target) -> Result<(Features, Features), String>;
+}
```

Comment:
This is probably not needed once we move the supported features into each of backend engines